src/Components/Component.php

```diff
@@ -203,4 +203,13 @@ protected function indent( $indent = 0 ) {
 
 		return "\n" . str_repeat( "\t", $this->mIndent );
 	}
+
+	protected function getAttribute( $attributeName, $default = null ) {
+		$attributeValue = $this->getDomElement()->getAttribute( $attributeName );
+		if( empty( $attributeValue ) ) {
```

Comment:
`empty` includes the empty string which might be what is actually specified in the XML. (Also `getAttribute` will return the empty string if the attribute is not present, so does not allow distinguishing between the attribute not being there and the attribute value being empty.)
Would it make sense to check for `$domElement->hasAttribute( $attributeName )` instead of `empty`?

Also: One more space after the `if `, please. :)